### PR TITLE
ci(): bug report form

### PR DESCRIPTION
**Pull request by ShaMan123**
### First merge #8189 

- Added a github action to have the user check some boxes before opening an issue
- Added a github action to automatically update the list of available versions for the form to choose from

https://github.com/fabricjs/fabric.js/compare/readme...bug-report-form

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -1,45 +1,56 @@
...
 ---
 
-<!-- ISSUES THAT ARE NOT BUGS OR LACK A TEST CASE WILL BE CLOSED. -->
+## Check List
 
-<!-- BUG TEMPLATE -->
+- [ ] I have read and followed the [Contributing Guide](/CONTRIBUTING.md)
+- [ ] I have read and followed the [Issue Tracker Guide](/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
+- [ ] I have searched and referenced existing issues and discussions
+- [ ] I have managed to reproduce the bug **after** upgrading to the **latest** version
```
Comment (asturur) [inline .github/ISSUE_TEMPLATE/bug_report.md:12]: or latest version of the same major version ?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+        - 0.6.7
+        - 0.6.6
+        - 0.6.4
+        - 0.5.13
+        - 0.5.10
+        - 0.5.7
+        - 0.5.6
+        - 0.5.5
+        - 0.5.3
+        - 0.5.2
```
Comment (asturur) [inline .github/ISSUE_TEMPLATE/bug_report.yml:305]: ugh. I would propose only the newest from every major is available, from 2.x up

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ Demos and examples 🤓 can be found on [fabricjs.com](http://fabricjs.com/demos
 ## 🏷️ Issue Tracker
 
 - **Before You Begin** 🎬
-  - 📌 Make sure you didn't fall into a known [**GOTCHA**](http://fabricjs.com/fabric-gotchas)
+  - 📌 Make sure you didn't fall into a known [**GOTCHA**](./GOTCHAS.md)
```
Comment (asturur) [inline CONTRIBUTING.md:19]: i would like to leave the link to the website. The money from the ads corner ( is not even a banner ) can be useful at some point

**Comment by asturur** [comment]
Did you put back all the doc pr here? i realized after reading half of it

**Comment by asturur** [comment]
oh it was in the description

File: .github/workflows/npmpublish.yml [hunk 4]
```diff
@@ -22,3 +22,17 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+    - run: npm ci
+    - run: node scripts/updateBugReportVersion.mjs
+      env:
+        VERSION: ${{ github.event.release.tag_name }}
+    - run: git commit -m "ci(): update bug report version" -a
```
Comment (asturur) [inline .github/workflows/npmpublish.yml:37]: i don't mind an automatic commit per release, till we release once in a while.
But if we release each small bugfix ( auspicable ) this becomes like the automatic changelog

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (ShaMan123) [reply .github/ISSUE_TEMPLATE/bug_report.yml:305]: It is generated by the action.
I can do a min version and from that onwards.
Good enough?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (ShaMan123) [reply .github/ISSUE_TEMPLATE/bug_report.md:12]: Why actually?
Do we want that?
Supporting old majors?
I don't

File: .github/workflows/npmpublish.yml (hunk 4 above)
Comment (ShaMan123) [reply .github/workflows/npmpublish.yml:37]: explain your concerns?
history pollution?
We can change this action to run chronologically instead, once a week or 2

Comment (ShaMan123) [reply .github/workflows/npmpublish.yml:37]: I will extract the action logic anyways to a standalone action

**Comment by ShaMan123** [comment]
I think I will close this PR now that I have authored an action for the form's version and merge the form into #8189

**Comment by ShaMan123** [comment]
ported to #8189 

todo
* gotchas link
*

**Pull request closed without merging**